PowerUp/app/src/main/java/powerup/systers/com/AvatarRoomActivity.java

```diff
@@ -270,6 +188,88 @@ public void onClick(View v) {
         getmDbHandler().close();
     }
 
+    @Override
+    protected void onSaveInstanceState(Bundle savedInstanceState) {
+        super.onSaveInstanceState(savedInstanceState);
+        savedInstanceState.putInt("eyeNum", eye);
+        savedInstanceState.putInt("faceNum", face);
+        savedInstanceState.putInt("hairNum", hair);
+        savedInstanceState.putInt("clothNum", cloth);
+    }
+
+    @Override
+    protected void onRestoreInstanceState(Bundle savedInstanceState) {
+        super.onRestoreInstanceState(savedInstanceState);
+        eye = savedInstanceState.getInt("eyeNum");
+        eye_setter(eye);
+        face = savedInstanceState.getInt("faceNum");
+        face_setter(face);
+        cloth = savedInstanceState.getInt("clothNum");
+        cloth_setter(cloth);
+        hair = savedInstanceState.getInt("hairNum");
+        hair_setter(hair);
+    }
+
+    public void eye_setter(int eye) {
+        String eyeImageName = getResources().getString(R.string.eye);
+        eyeImageName = eyeImageName + eye;
+        R.drawable ourRID = new R.drawable();
+        java.lang.reflect.Field photoNameField;
+        try {
+            photoNameField = ourRID.getClass().getField(eyeImageName);
+            eyeView.setImageResource(photoNameField.getInt(ourRID));
+            eyeAvatar.setImageResource(photoNameField.getInt(ourRID));
+        } catch (NoSuchFieldException | IllegalAccessException
+                | IllegalArgumentException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void hair_setter(int hair) {
+        String hairImageName = getResources().getString(R.string.hair);
+        hairImageName = hairImageName + hair;
+        R.drawable ourRID = new R.drawable();
+        java.lang.reflect.Field photoNameField;
+        try {
+            photoNameField = ourRID.getClass().getField(hairImageName);
+            hairView.setImageResource(photoNameField.getInt(ourRID));
+            hairAvatar.setImageResource(photoNameField.getInt(ourRID));
+        } catch (NoSuchFieldException | IllegalAccessException
+                | IllegalArgumentException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public void cloth_setter(int cloth) {
```

Comment:
@chhavip I will try to combine them in a single function. Can you give me 1-2 days for it since currently I am working on a issue of first-aid app.
_Thank you_